framework/tlvf/tlvf.py

```diff
@@ -710,8 +710,9 @@ def addClassParam(self, obj_meta, param_name, param_type, param_type_info, param
                             param_val_const = param_type + "::" + param_val_const
                         lines_cpp.append("if (!m_%s__) *m_%s = %s;" % ( self.MEMBER_PARSE, param_name, param_val_const) )
                     elif param_val != None: lines_cpp.append("if (!m_%s__) *m_%s = %s;" % ( self.MEMBER_PARSE, param_name, param_val) )
-                    elif param_length_var:  lines_cpp.append("if (!m_%s__) *m_%s = 0;" % ( self.MEMBER_PARSE, param_name) )
-                    lines_cpp.append( "m_%s__ += sizeof(%s) * 1;" % ( self.MEMBER_BUFF_PTR, param_type) )
+                    elif param_length_var: lines_cpp.append("if (!m_%s__) *m_%s = 0;" % ( self.MEMBER_PARSE, param_name) )
+                    lines_cpp.append("%sif (!buffPtrIncrementSafe(sizeof(%s))) { return false; }" % (self.getIndentation(0), param_type) )
```

Comment:
There's another increment on line 810. I'll admit that there, you can statically determine that it will not overflow because the offset is exactly equal to the remaining bytes, but still it's better IMO to do it consistently.

There's another increment on line 839 and on line 841, and those *are* essential I think. In fact, I think that that is the one that triggered the issue to begin with.

There are increments on line 1053 and 1081, but again those are not overflowing because of conditions that come before. There, however, I think it would make sense to use a condition on buffPtrIncrementSafe instead.

And finally, I think the condition on line 1434 can be removed, as it is essentially checking the same thing, only too late because the out-of-bounds accesses have already happened.